pkg/api/controller.go

```diff
@@ -3006,6 +3011,57 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email string, token string) error {
+	return c.Emailer.SendEmail([]string{email}, token, token, nil)
+}
+
+func (c *Controller) PasswordForgot(w http.ResponseWriter, r *http.Request, body PasswordForgotJSONRequestBody) {
+	user, err := c.Auth.GetUserByEmail(r.Context(), body.Email)
```

Comment:
We need to verify the request input (email), before you call get user by email.
You can use the following before calling the lookup:
```go
addr, err := mail.ParseAddress(body.Email)
if err != nil {
    c.Logger.WithError(err).WithField("email", body.Email).Debug("forgot password with invalid email")
    w.WriteHeader(http.StatusNoContent)
    return
}
```